datanode/dehistory/fsutil/fsutil.go

```diff
@@ -0,0 +1,203 @@
+package fsutil
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	vgfs "code.vegaprotocol.io/vega/libs/fs"
+)
+
+// TODO File containing common file manipulation logic used across dehistory - exists elsewhere?
+
+// RemoveAllFromDirectory used in place of os.RemoveAll when the directory should be emptied but not removed.
+func RemoveAllFromDirectory(dir string) error {
```

Comment:
Using https://pkg.go.dev/path/filepath#WalkDir here would probably replace most of this function content.